malariagen_data/anoph/sample_metadata.py

```diff
@@ -507,18 +521,19 @@ def sample_metadata(
         for on, data in self._extra_metadata:
             df_samples = df_samples.merge(data, how="left", on=on)
 
-        # For convenience, apply a query.
+        # For convenience, apply a sample selection.
         if sample_query is not None:
-            if isinstance(sample_query, str):
-                # Assume a pandas query string.
-                df_samples = df_samples.query(sample_query)
-            else:
-                # Assume it is an indexer.
-                df_samples = df_samples.iloc[sample_query]
+            # Assume a pandas query string.
+            df_samples = df_samples.query(sample_query)
+            df_samples = df_samples.reset_index(drop=True)
+        elif sample_indices is not None:
+            # Assume it is an indexer.
+            df_samples = df_samples.iloc[sample_indices]
             df_samples = df_samples.reset_index(drop=True)
```

Comment:
Example of how we now use two parameters `sample_query` and `sample_indices` to more clearly handle different types of sample selection.